Factory.Cargo/CargoFactoryApi.cs

```diff
@@ -0,0 +1,36 @@
+using Factory.Cargo.Factory;
+using Factory.Cargo.TransportationFactory;
+
+namespace Factory.Cargo;
+
+public class CargoFactoryApi
+{
+    private Dictionary<string, ITransportationFactory> _transportationFactories = new();
+
+    public void AddBaseFactories()
+    {
+        _transportationFactories.Add("Air",new AirTransportationFactory());
+        _transportationFactories.Add("Train",new TrainTransportationFactory());
+        _transportationFactories.Add("Ship",new ShipTransportationFactory());
+    }
+    
+    
+    public void AddTransportationFactory(string mode, ITransportationFactory transportationFactory)
+        => _transportationFactories.Add(mode, transportationFactory);
+
+    public ITransportation Create(string mode, string? origin = null, string? destination = null)
+    {
+        if (_transportationFactories.TryGetValue(mode, out var transportationFactory) || transportationFactory is null)
+            throw new NotImplementedException($"The factory {mode} didn't define.");
+        try
+        {
+            return transportationFactory.Create(origin, destination);
```

Comment:
if you have 10 differ factory. how to call them? 

in this case you are calling in a try catch and then call another in line 34.

Reply:
@mohammadKarimi 
this issue addressed on previous review comment.